src/character.cpp

```diff
@@ -8910,7 +8910,9 @@ bool Character::dispose_item( item_location &&obj, const std::string &prompt )
                        e.prompt, e.moves ) );
     }
 
-    menu.query();
+    do {
+        menu.query();
+    } while( menu.ret == UILIST_CANCEL );
```

Comment:
```suggestion
    menu.allow_cancel = false;
    menu.query();
```
`uilist` has a flag for this purpose.

Reply:
That is good to know, thank you!